01_fib/ailyntang/fib_test.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("1\n1\n2\n3\n5\n8\n13\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output from main()")
+		t.Errorf("expected: %v\nactual: %v", expected, actual)
+	}
+}
+
+func TestUnexpectedInputs(t *testing.T) {
```

Comment:
:+1: I like that you've split up the tests for expected and unexpected (error) inputs. It helps make the code being tested more unstandable by more clearly showing the domain of the function.

Reply:
Thanks for letting me know that's a helpful thing to do. I'll continue doing this in the future.